src/python/WMCore/WMRuntime/Tools/Scram.py

```diff
@@ -102,6 +102,43 @@ def isCMSSWSupported(thisCMSSW, supportedCMSSW):
 
     return False
 
+def isEnforceGUIDInFileNameSupported(thisCMSSW):
+    """
+        _isEnforceGUIDInFileNameSupported_
+
+        Function used to validate whether the CMSSW release to be used supports
+        the enforceGUIDInFileName feature.
+        :param thisCMSSW: release to be used in this job
+
+    """
+    # a set of CMSSW releases that support the enforceGUIDInFileName feature. Releases in the same
+    # cycle with a higher minor revision number also support the feature.
+    supportedReleases = set(["CMSSW_10_6_8", "CMSSW_10_2_20", "CMSSW_9_4_16", "CMSSW_9_3_17", "CMSSW_8_0_34"])
+    # a set of specific CMSSW releases that supported the enforceGUIDInFileName feature.
+    specificSupportedReleases = set(["CMSSW_10_2_20_UL", "CMSSW_9_4_16_UL", "CMSSW_8_0_34_UL", "CMSSW_7_1_45_patch3"])
+
+    if not thisCMSSW:
+        logging.info("You must provide the CMSSW version being used by this job.")
+        return False
+
+    # true if CMSSW release is >= CMSSW_11_0_0
+    if isCMSSWSupported(thisCMSSW, "CMSSW_11_0_0"):
+        return True
+    # true if CMSSW release is in the specific release set
+    elif thisCMSSW in specificSupportedReleases:
+        return True
+    # true if the CMSSW release's minor revision is >= to one of the supported releases
+    else:
+        thisMajor, thisMid, thisMinor = [int(i) for i in thisCMSSW.split('_', 4)[1:4]]
+        for release in supportedReleases:
+            supportedMajor, supportedMid, supportedMinor = [int(i) for i in release.split('_', 4)[1:4]]
+            # major and mid revisions need an exact match
+            if thisMajor  == supportedMajor and thisMid == supportedMid:
+                # minor revision need to be >= to the supported minor revision
+                if thisMinor >= supportedMinor:
```

Comment:
This is a tricky one, and I'd rather be safe and only evaluates to True if it's greater than.
For instance, `CMSSW_7_1_45_patch2` would fall into this `else` block and would evaluate to True (because we drop the last part of the release name)

Reply:
Since `CMSSW_7_1_45` isn't in `supportedReleases` I don't think the case you mention will be a problem. 
https://github.com/dmwm/WMCore/blob/57e3fdd5031e9ece220a4c58848227f2d8b480de/src/python/WMCore/WMRuntime/Tools/Scram.py#L114-L116
The exact matches for supported versions are handled by:
https://github.com/dmwm/WMCore/blob/57e3fdd5031e9ece220a4c58848227f2d8b480de/src/python/WMCore/WMRuntime/Tools/Scram.py#L117-L118

We aren't supporting anything else in `CMSSW_7_1_X` except the specific `CMSSW_7_1_45_patch3` version, correct?

Reply:
In this L138, you actually implemented support for minor releases. E.g., CMSSW_7_1_45_patch3 would be True, CMSSW_7_1_45_patch4 would be true, CMSSW_7_1_45_patch2 would be true and anything beyond that, for instance CMSSW_7_1_46, CMSSW_7_1_47, CMSSW_7_1_48... 

And this is the expected behavior, besides one subtle detail in the >= vs >. If we use >=, then we could be enabling that feature for releases that do not support it, e.g.: CMSSW_7_1_45_patch2 or CMSSW_7_1_45.